lib/utils/fs-utils.js

```diff
@@ -59,8 +59,53 @@ function existsAndNonZero(path) {
         });
 }
 
+function isSameFlavorSet(flavorsPath, flavorsSet, excludeDirs, pattern) {
+    return qio.listTree(flavorsPath,
+        (path, stat) => {
+            // read recording content path and return the flavors dirs
+            if (!stat.isDirectory()) {
+                return false;
+            }
+            let dirName = path.split("/").pop();
+
+            if (excludeDirs.includes(dirName)) {
+                return false;
+            }
+            if (pattern.test(dirName)) {
+                return true;
+            } else {
+                return false;
+            }
+        })
+        .then((flavorsDirs)=> {
+            let flavorNames = _.map(flavorsDirs, function(flavorDir) {
```

Comment:
the naming is confusing... flavorNames should be called currentFlavors . or previousFlavors